velox/connectors/hive/HiveConnectorUtil.cpp

```diff
@@ -582,18 +582,24 @@ bool testFilters(
   for (const auto& child : scanSpec->children()) {
     if (child->filter()) {
       const auto& name = child->fieldName();
-      if (!rowType->containsChild(name)) {
-        // If missing column is partition key.
-        auto iter = partitionKey.find(name);
+      auto iter = partitionKey.find(name);
+      // The partition key columns are writen in the data file for
+      // IcebergTables, so we need to test both cases
+      if (!rowType->containsChild(name) || iter != partitionKey.end()) {
```

Comment:
@yingsu00 Will this work if we remove `hive.parquet.use-column-names=true` in Presto?